uitools/cpp/Esri/ArcGISRuntime/Toolkit/UtilityNetworkTraceController.cpp

```diff
@@ -0,0 +1,1026 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "UtilityNetworkTraceController.h"
+
+// ArcGISRuntime headers
+#include <ArcGISFeature.h>
+#include <ArcGISFeatureListModel.h>
+#include <AttributeListModel.h>
+#include <Error.h>
+#include <FeatureLayer.h>
+#include <FeatureTable.h>
+#include <GeoView.h>
+#include <GeometryEngine.h>
+#include <Graphic.h>
+#include <GraphicListModel.h>
+#include <GraphicsOverlay.h>
+#include <GraphicsOverlayListModel.h>
+#include <IdentifyLayerResult.h>
+#include <Map.h>
+#include <Multipoint.h>
+#include <Polygon.h>
+#include <Polyline.h>
+#include <Renderer.h>
+#include <SimpleFillSymbol.h>
+#include <SimpleLineSymbol.h>
+#include <SimpleMarkerSymbol.h>
+#include <SpatialReference.h>
+#include <Symbol.h>
+#include <SymbolTypes.h>
+#include <TaskWatcher.h>
+#include <UtilityAssetGroup.h>
+#include <UtilityAssetType.h>
+#include <UtilityElement.h>
+#include <UtilityElementTraceResult.h>
+#include <UtilityFunctionTraceResult.h>
+#include <UtilityGeometryTraceResult.h>
+#include <UtilityNamedTraceConfiguration.h>
+#include <UtilityNetwork.h>
+#include <UtilityNetworkAttribute.h>
+#include <UtilityNetworkListModel.h>
+#include <UtilityNetworkSource.h>
+#include <UtilityNetworkTypes.h>
+#include <UtilityTerminalConfiguration.h>
+#include <UtilityTraceFunction.h>
+#include <UtilityTraceFunctionOutput.h>
+#include <UtilityTraceParameters.h>
+#include <UtilityTraceResult.h>
+#include <UtilityTraceResultListModel.h>
+#include <Viewpoint.h>
+
+// Toolkit headers
+#include "Internal/DisconnectOnSignal.h"
+#include "Internal/DoOnLoad.h"
+#include "Internal/GeoViews.h"
+#include "UtilityNetworkFunctionTraceResult.h"
+#include "UtilityNetworkFunctionTraceResultsModel.h"
+#include "UtilityNetworkListItem.h"
+#include "UtilityNetworkTraceStartingPoint.h"
+#include "UtilityNetworkTraceStartingPointsModel.h"
+
+// Qt headers
+#include <QList>
+#include <QUuid>
+
+// std headers
+#include <cmath>
+
+namespace Esri::ArcGISRuntime::Toolkit {
+
```

Comment:
needs class level doc. the controllers are public types that we generally doc

Reply:
👍